test/cqlpy/test_secondary_index.py

```diff
@@ -1956,3 +1956,16 @@ def test_paging_and_drop_index_no_allow_filtering(cql, test_keyspace):
                 assert len(r.current_rows) <= page_size
                 got.extend(r.current_rows)
             assert expected == got
+
+
+# Test index representation in system.* tables
+def test_index_in_system_tables(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as table:
+        index_name = unique_name()
+        cql.execute(f"CREATE INDEX {index_name} ON {table}(v)")
+        res = [ f'{r.keyspace_name}.{r.view_name}' for r in cql.execute('select * from system.built_views')]
+        assert f'{test_keyspace}.{index_name}_index' in res
+        res = [ f'{r.table_name}::{r.index_name}' for r in cql.execute('select * from system."IndexInfo"')]
```

Comment:
I didn't understand the discussion in the commit message - this test does pass with Cassandra, right? Perhaps what you meant to say in this commit message was that this test would have caught https://github.com/scylladb/scylladb/issues/8600 if we had this test?

Reply:
> I didn't understand the discussion in the commit message - this test does pass with Cassandra, right?

It now should, I only ran it with scylla

> Perhaps what you meant to say in this commit message was that this test would have caught https://github.com/scylladb/scylladb/issues/8600 if we had this test?

Yes

Reply:
I see. Thanks. Please do try to test it in cassandra -
`test/cqlpy/run-cassandra test_secondary_index.py::testname`.

Reply:
C* doesn't put index into built_views, other than that 

```
$ CASSANDRA=/home/xemul/src/cassandra/apache-cassandra-5.0.2/bin/cassandra ./test/cqlpy/run-cassandra test_secondary_index.py::test_index_in_system_tables
Cassandra is: /home/xemul/src/cassandra/apache-cassandra-5.0.2/bin/cassandra.
Booting Cassandra on 127.1.85.215 in /tmp/scylla-test-21975...
JAVA_HOME: /usr/lib/jvm/jre-11
Boot successful (24.0 seconds).
======== test session starts =============================================================================================================
platform linux -- Python 3.12.7, pytest-7.2.2, pluggy-1.5.0
rootdir: /home/xemul/src/scylla/test/cqlpy, configfile: pytest.ini
plugins: asyncio-0.20.3, allure-pytest-2.13.5
asyncio: mode=Mode.STRICT
collected 1 item                                                                                                                                                                                                                             

test_secondary_index.py .                                                                                                                                                                                                              [100%]

<skip warnings>
======= 1 passed, 4206 warnings in 2.46s ======================================================================================================
```

Reply:
Indeed, Cassandra's secondary index are NOT views at all, they are implemented differently.
So I you either need to split the "secondary index gets listed as a view" part of the test to a separate test function that is "scylla_only", or just get rid of this check - just check that a view gets listed as a view and an index gets listed as an index - but don't try to check that an index gets listed as a view.

Reply:
I did it like this:

```
        cql.execute(f"CREATE INDEX ...")
        if is_scylla(cql):
            test system.build_views table
        test system.IndexInfo table
        test system.IndexInfo table with WHERE selection
```

Reply:
Ok, I personally prefer to have separate small test - one small test passes on both Scylla and Cassandra and one small test passes on just Scylla, with an oportunity to explain in a comment why this is the case.
Also I'm not sure why you want to test in this test the fact that Scylla's index is implemented as a view and what exactly is its name, it sounds like a different test.

In any case, this is a minor detail - so what you plan to do is fine too.